File: src/main/java/com/vinyl/model/Track.java
package com.vinyl.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Track {
	private String trackCatalogNum;
	private String trackName;
	private Integer duration;
	private List<String> composerIds;
	private List<String> albumIds;
	private List<String> artistIds;
	private List<String> bandIds;
	private List<String> featuringArtistIds;
	private List<String> featuringBandIds;
	private List<String> languages;
}

File: src/main/resources/frontend/src/schemas/track.js
export default {
  props: {
    trackCatalogNum: 'int',
    trackName: 'string',
    duration: 'int',
    composerIds: {
      type: 'composer',
      isSchema: true,
      isArray: true
    },
    /*"albumIds: [
      "101"
    ],*/
    artistIds: {
      type: 'artist',
      isSchema: true,
      isArray: true,
      readonly: true
    },
    bandIds: {
      type: 'band',
      isSchema: true,
      isArray: true,
      readonly: true
    },
    featuringArtistIds: {
      type: 'artist',
      isSchema: true,
      isArray: true,
      readonly: true
    },
    featuringBandIds: {
      type: 'band',
      isSchema: true,
      isArray: true,
      readonly: true
    },
    languages: {
      type: 'country',
      isArray: true,
      joins: ['track_language']
    },
    artistBindings: {
      type: 'artist2track',
      isSchema: true,
      isArray: true,
      filterOnly: true
    }
  },
  key: 'trackCatalogNum',
  display: (item) => item.trackName
}